### ISSUE TITLE
Default false value not setting correctly on create

### ISSUE THREAD

**Issue by andrewyoo** [composite-primary-keys/composite_primary_keys#518]
I am experiencing an issue where I run `Model.create` with a column in the composite index that is defaulted to `false`.  The object returned has this field set to `nil`. The record is correctly created, but the object instantiated is not.  

This is occurring in v11.3.1

I have tracked the change to this:
 
https://github.com/composite-primary-keys/composite_primary_keys/commit/e7dab1432747c0ade3ca13c36d6f7ac8069418e2

I was wondering why this change was made to merge the changes together.  I believe this is in error since `(false || nil)` would return `nil`. 

Ruby: 2.6.6
Rails: 5.2.4.2
CPK: 11.3.1
Mysql: 5.7

**Comment by cfis** [comment]
The reason for that change was to allow setting some of the keys and filling the remaining ones with default values (there was a issue about it IIRC).

In general I think that's the right approach. But I didn't consider that a valid value would be false. Want to add a patch for that case?

**Comment by andrewyoo** [comment]
In older versions (11.2.0 and the rails 4.2 version) i didn't have an issue with ActiveRecord not correctly setting the defaulted values, but I can only say that was the case for mysql. I have submitted a PR that you can look at.

Thanks!

**Comment by ral-wtag** [comment]
@cfis  I have faced this issue in the latest version. 
While I was trying to create an object which has cpk (e.g `User.create()`) it creates the object in the database but reruns id nil.

I have tried to solve the issue by overriding the [_create_record](https://github.com/composite-primary-keys/composite_primary_keys/blob/master/lib/composite_primary_keys/persistence.rb#L68-L70) method like this, and it seemed to work

I removed the condition `self.id.compact.empty?` as the id was giving me `[nil, 1]`(user-id nil) and this condition was false for this, but it needs to be in this block to set the id I guess

```
if self.composite?
    self.id = self.id.zip(Array(new_id)).map {|id1, id2| (id1 || id2)}
else
    self.id ||= new_id if self.class.primary_key
end
```

**Comment by cfis** [comment]
Ok, this should be fixed by cf82cef70a7b92705dce5b5302c0f742408be794.
